narwhals/_duckdb/dataframe.py

```diff
@@ -142,7 +142,14 @@ def drop(self: Self, columns: list[str], strict: bool) -> Self:  # noqa: FBT001
         selection = (col for col in self.columns if col not in columns_to_drop)
         return self._from_native_frame(self._native_frame.select(*selection))
 
-    def lazy(self: Self) -> Self:
+    def lazy(self: Self, *, backend: Implementation | None = None) -> Self:
+        # The `backend`` argument has no effect but we keep it here for
+        # backwards compatibility because in `narwhals.stable.v1`
+        # function `.from_native()` will return a DataFrame for DuckDB.
+
+        if backend is not None:  # pragma: no cover
```

Comment:
I think duckdb should be allowed?

```suggestion
        if backend not in (None, Implementation.DUCKDB):  # pragma: no cover
```

Reply:
i don't think it needs to be - if this path only exists for v1 backwards-compatibility, then https://github.com/narwhals-dev/narwhals/pull/1895#discussion_r1936126615 is something somebody could have written, but they wouldn't have passed a backend, so raising for `backend is not None` wouldn't break anyone's code